filewatcher/filewatcher.go

```diff
@@ -58,12 +64,28 @@ const (
 // Inconsistent type will cause panic, as does returning nil data and nil error.
 type Parser func(f io.Reader) (data interface{}, err error)
 
+// DirParser is a callback function to be called when a file in a watched directory
+// has been touched, or read for the first time.
+// Should always return a constant type or will cause panic
+type DirParser func(path string) (data interface{}, err error)
+
+// DirParserToParser wraps a DirParser so that it may be used as a Parser in a
+// filewatcher
+func DirParserToParser(dp DirParser) Parser {
```

Comment:
I'm bad at names and I was just using `DirParserToParser` as a placeholder in my previous comment, not that it's a good name 😅 . maybe `DirParserWrapper` or something is better?